### ISSUE TITLE
insights: backend to support different time intervals based on insight configuration

### ISSUE THREAD

**Issue by coury-clark** [sourcegraph/sourcegraph#21778]
The code insights backend currently operates on a fixed two week interval, but the extension driven code insights allows users to define a custom range and interval
```json
"searchInsights.insight.telemetryService": {
    "title": "Telemetry service",
    "repositories": [
      "github.com/sourcegraph/sourcegraph"
    ],
    "series": [
      {
        "name": "testing insight personal",
        "stroke": "var(--oc-grape-7)",
        "query": "telemetryService.log"
      }
    ],
    "step": {
      "months": 2
    }
  }
```

**Comment by felixfbecker** [comment]
Note: It would be an acceptable regression (for moving the frontend to the backend, and even for Beta) if the backend insights just used a hard-coded X range of 1 year and we removed/commented out the setting in the frontend creation UI until we can support it again.

**Comment by coury-clark** [comment]
This has been implemented and released in 3.35
